stage-audio/03-hardware/files/ovos-i2csound

```diff
@@ -15,190 +15,183 @@
-    if [ "$SJ201LED" ] ; then
-       echo "Found revision-6 SJ-201 board"
-       # Initializing and resetting LED ring
-       /usr/bin/sj201-reset-led
-       # Reset FAN to low speed
-       /usr/sbin/i2cset -a -y 1 0x04 101 30 i
-    else
-       echo "Assume revision-10 SJ-201 board"
-       dtoverlay sj201-rev10-pwm-fan-overlay
+}
+
+# Removes existing configuration files safely
+remove_config_files() {
+    rm -f "/etc/pulse/system.pa"
+    rm -f "/etc/pulse/default.pa"
+    rm -f "/etc/pulse/daemon.conf"
+    rm -f "/var/lib/alsa/asound.state"
+    rm -f "/etc/ovos_asound.state"
+}
+
+# Creates necessary symbolic links for configuration files
+create_configuration_symlinks() {
+    ln -sf "$ASOUND_STATE" /etc/ovos_asound.state
+    ln -sf "$ASOUND_STATE" /var/lib/alsa/asound.state
+    ln -sf "$PULSE_SYSTEM" /etc/pulse/system.pa
+    ln -sf "$PULSE_SYSTEM" /etc/pulse/default.pa
+    ln -sf "$PULSE_DAEMON" /etc/pulse/daemon.conf
+}
+
+# Main execution
+main() {
+    # Detecting I2C devices
+    for device in "${!devices[@]}"; do
+        address="${devices[$device]}"
+        if detect_i2c_device "$address"; then
+            detection_results[$device]=true
+        else
+            detection_results[$device]=false
+        fi
+        echo "$device detection result: ${detection_results[$device]}"
+    done
+
+    # Handling hardware-specific configurations
+    if [[ ${detection_results[WM8960]} == true ]] ; then
+        echo "WM8xxx based HAT found"
+        atmega328p=$(avrdude -p atmega328p -C /etc/avrdude-gpio.conf -c linuxgpio -U signature:r:-:i -F 2>/dev/null | head -n1)
+        if [ "${atmega328p}" == ":030000001E950F3B" ] ; then
+            detection_results[MARK1]=true
+            echo "Mark-1 enclosure $MARK1"
+        else
+            echo "WM8960 based 2-mic $WM8960"
+        fi
     fi
-    echo "Configuring buttons"
-    dtoverlay sj201-buttons-overlay
-#
-    PULSE_SYSTEM=/etc/pulse/default.pa.d/sj201-default.pa
-    PULSE_DAEMON=/usr/share/pulseaudio/alsa-mixer/profile-sets/sj201-daemon.conf
-fi
-
-if [ "$AIYVOICEBONNET" ] ; then
-    echo "AIY VoiceBonnet"
-#     ASOUND_STATE=/usr/share/pulseaudio/alsa-mixer/profile-sets/aiy-voicebonnet-v2.conf
-fi
-
-# Install soundstate
-echo "create asound status file"
-ln -s $ASOUND_STATE /etc/ovos_asound.state
-ln -s $ASOUND_STATE /var/lib/alsa/asound.state
-#
-# Install pulseaudio files
-echo "create pulse system file"
-ln -s $PULSE_SYSTEM /etc/pulse/system.pa
-echo "create pulse default file"
-ln -s $PULSE_SYSTEM /etc/pulse/default.pa
-echo "create pulse daemon file"
-ln -s $PULSE_DAEMON /etc/pulse/daemon.conf
-#
-# # Restore sound levels
-alsactl -E HOME=/run/alsa -f /etc/ovos_asound.state restore
+
+    if [[ ${detection_results[WM8960]} == true ]] && [[ ${detection_results[RESPEAKER4]} == false ]] ; then
+        echo "Installing and configuring WM8960 based 2-mic HAT"
+        dtoverlay wm8960-soundcard
+        ASOUND_STATE=/etc/voicecard/wm8960_asound.state
```

Comment:
This file is not here by default.  I had it to where the amixer commands would be done, and then stored to the `.state` file. and loaded at the end.  Maybe not the best way.  I do know that the amixer controls do the same thing as a state file.

Reply:
I have refactored the code, and left the original logic in this file.
Please tell me what change should I make here?
I see that the state file is present but it differs with the configuration amixer

Reply:
With the file from https://github.com/OpenVoiceOS/ovos-i2csound/blob/dev/ovos-i2csound, the state files are no longer needed. 

They need removed in `run.sh` also

If you split this pr, I will merge the others, and help work on i2csound

Reply:
done, file reverted